src/framework/CMakeLists.txt

```
@@ -76,6 +76,17 @@ add_custom_target("Other_mlt_Files" SOURCES
 
 add_library(Mlt${MLT_VERSION_MAJOR}::mlt ALIAS mlt)
 
+# Generate export header for cross-platform symbol visibility
+include(GenerateExportHeader)
+generate_export_header(mlt
+  BASE_NAME mlt
+  EXPORT_MACRO_NAME MLT_API
+  EXPORT_FILE_NAME mlt_export.h
+  DEPRECATED_MACRO_NAME MLT_DEPRECATED
+  NO_EXPORT_MACRO_NAME MLT_NO_EXPORT
+  STATIC_DEFINE MLT_STATIC_DEFINE
+)
```

No need to overwrite the defaults with default values:

```suggestion
generate_export_header(mlt
  EXPORT_MACRO_NAME MLT_API
)
```

@ddennedy @bmatherly Any preference on the naming `MLT_EXPORT` (CMake's default) vs. `MLT_API`

---

No preference for me

---

I can tell you now that export is more common, but it is a bike shed topic.

---

Thinking more about this and backed by the new findings regarding export headers for every module I think it would be better to use `MLT_EXPORT` instead of `MLT_API`. @smiller would you mind to change it?

---

I agree if adding calls to `generate_export_header` for each module fixes it because it will make the names more consistent, less cmake lines, more clear.